transport/peer/hostport/peer.go

```diff
@@ -0,0 +1,73 @@
+package hostport
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new hostport.PeerIdentifier
+func NewPeerIdentifier(hostport string) PeerIdentifier {
+	return PeerIdentifier(hostport)
+}
+
+// PeerIdentifier uniquely references a host:port combination using a common interface
+type PeerIdentifier string
+
+// Identifier generates a (should be) unique identifier for this PeerIdentifier (to use in maps, etc)
+func (p PeerIdentifier) Identifier() string {
+	return string(p)
+}
+
+// NewPeer creates a new hostport.Peer from a hostport.PeerIdentifier, transport.Agent, and transport.PeerSubscriber
+func NewPeer(pid PeerIdentifier, agent transport.PeerAgent, subscriber transport.PeerSubscriber) *Peer {
+	return &Peer{
+		PeerIdentifier:   pid,
+		agent:            agent,
+		subscriber:       subscriber,
+		connectionStatus: transport.PeerAvailable,
+	}
+}
+
+// Peer keeps a subscriber to send status updates to it, and the PeerAgent that created it
+type Peer struct {
+	PeerIdentifier
+
+	agent            transport.PeerAgent
+	subscriber       transport.PeerSubscriber
```

Comment:
Peers will have multiple subscribers. This should be a slice, and notifications should be broadcast.